metrics/metrics.go

```diff
@@ -135,7 +136,13 @@ func appendDefaults(prefix string) (string, error) {
 }
 
 // stubbed out for testing
-var hostname = os.Hostname
+var hostname = func() (string, error) {
+	out, err := exec.Command("hostname", "-f").Output()
```

Comment:
maybe https://github.com/ShowMax/go-fqdn/blob/master/fqdn.go is better solution ?